build/github/cockroach-microbench-ci.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+# Directories and file names
+output_dir="./artifacts/microbench"
+cleaned_current_dir="$output_dir/current"
+cleaned_base_dir="$output_dir/base"
+
+# hardcoded the file name as of now since `compare` requires a particular format for the file name.
+benchmark_file_name="pkg_sql-report.log"
+log_output_file_path="$output_dir/microbench.log"
+storage_bucket_url="gs://cockroach-microbench-ci"
+
+# Threshold for comparison
+threshold=${COMPARE_THRESHOLD:-0}
+success_exit_status=0
+error_exit_status=1
+
+# Exit early if SKIP_COMPARISON is true and not a push step
+if $SKIP_COMPARISON && ! $PUSH_STEP; then
+  echo "Exiting since skip comparison is enabled and this is not a push step."
+  exit $success_exit_status
+fi
+
+# Build binary with Bazel
+bazel build --config crosslinux $(./build/github/engflow-args.sh) --jobs 100 //pkg/cmd/roachprod-microbench
+
+roachprod_microbench_dir="_bazel/bin/pkg/cmd/roachprod-microbench/roachprod-microbench_"
+
+mkdir -p "$output_dir"
+
+# Run benchmarks and clean output
+# running count=4 here because that's the minimum required for a comparison
+bazel test //pkg/sql/tests:all \
+  --config=use_ci_timeouts \
```

Comment:
Please use EngFlow for the build  (`--config crosslinux $(./build/github/engflow-args.sh) --jobs 100`). Also add `--strategy=TestRunner=sandboxed` so the test runs locally instead of on the cluster.

Reply:
(I assume you want the tests to run locally as they're benchmarks)